added swagger-PS-feature

serhii-shyian opened the pull request.

serhii-shyian committed:
added swagger-PS-feature

Elena-Bruyako commented on src/main/java/com/example/bookstore/controller/BookController.java:
```diff
@@ -20,43 +23,56 @@
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Get all books by parameters",
+            description = "Get a list of all books according to the parameters")
     public List<BookDto> searchBooks(BookSearchParametersDto searchParameters) {
```
Add `@ParameterObject @PageableDefault Pageable pageable`

Elena-Bruyako commented on src/main/java/com/example/bookstore/controller/BookController.java:
```diff
@@ -20,43 +23,56 @@
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Get all books",
+            description = "Get a list of all available books")
+    public List<BookDto> getAll(Pageable pageable) {
```
```suggestion
    public List<BookDto> getAll(@ParameterObject @PageableDefault Pageable pageable) {
```

Elena-Bruyako commented on src/main/java/com/example/bookstore/model/Book.java:
```diff
@@ -37,5 +38,6 @@ public class Book {
     @Column(nullable = false)
-    private boolean isDeleted = false;
+    @Value("false")
+    private boolean isDeleted;
```
Remove `@NoArgsConstructor`

Elena-Bruyako reviewed (changes requested):
see pr

serhii-shyian committed:
1) Remove @NoArgsConstructor from Book model
2) Modify searchByParameters(BookSearchParametersDto params, Pageable pageable)
3) Modify searchBooks() and getAll() endpoints
